data/base/endpoint.php

```diff
@@ -104,28 +130,24 @@ protected function register() {
 	 * @param string $route
 	 * @param string $endpoint_class
 	 *
-	 * @return \Elementor\Data\Base\SubEndpoint
-	 * @throws \Exception
+	 * @return \Elementor\Data\Base\SubEndpoint|false
 	 */
-	protected function register_sub_endpoint( $route, $endpoint_class ) {
-		$endpoint_instance = new $endpoint_class( $route, $this );
+	public function register_sub_endpoint( $route, $endpoint_class ) {
+		$endpoint_instance = new $endpoint_class( $this, $route );
 
 		if ( ! ( $endpoint_instance instanceof SubEndpoint ) ) {
-			throw new \Exception( 'Invalid endpoint instance.' );
+			trigger_error( 'Invalid endpoint instance.' );
```

Comment:
Move to arg type.

Reply:
Handled: https://github.com/elementor/elementor/pull/12820/commits/9537cf2b915a0b508619636353fa886543fa01c5